[nichyjt] iP

nichyjt opened the pull request.

nichyjt committed:
Add Greet, Exit and PrintBorder functions for skeletal control flow

nichyjt committed:
Add input listener, Echo input and Exit features

nichyjt committed:
Add ability to create tasks and list tasks. Additional keywords for termination: "bye", "exit" added for ease of use

nichyjt committed:
Add new command handlers - <mark> and <unmark> tasks. Refactor all commands to run under a switch case. Refactor tasklist to use a Task class instead of a String primitive.

nichyjt committed:
Add Task class to abstract Task details

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Format Duke and Task files to adhere to coding standards

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Format Duke and Task files to adhere to coding standards

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add TODO, EVENT and DEADLINE subclasses

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Refactor command parsing logic to follow a <SUBCOMMAND, COMMAND_TEXT> paradigm. Add event, deadline, todo features and various method handles to help unwrap CLI input into the subcommand framework. Add input guard clauses.

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add TextUiTesting Expected output and input commands

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Format code to coding standards. Add guard clauses for <mark> <unmark> commands, ensure that they properly return an expected value.

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

GohJW commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,225 @@
+            newTask = new Deadline(description, deadline);
+        } else if (taskType.equals("event")) {
+
+            if (commandMap.size() < 3) {
```
Perhaps the use of named constants such as EVENT_COMMAND_LENGTH rather than magic numbers such as 2 or 3 for better understanding on what you are comparing to.

nichyjt replied:
Thank you for the suggestion! This is a good point and I think it'll be worth refactoring many of these 'magic numbers' into enum consts.

GohJW commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,225 @@
+        // For example: deadline my_task --by -xyz sunday-monday
+        // Will output: ["deadline my_task", "by -xyz sunday-monday"]
+        // For now, this does not account for accidental or adversarial '--' input at the start of words
+        // but this is an unlikely edge case that can be dealt with later
```
Good use of indentation on your comments to indicate which part of code you are specifying.

GohJW commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,225 @@
+        Duke.isListening = keepListening;
+    }
+
+    public static void PrintBorder() {
```
Method should be in camelCase printBorder()

nichyjt replied:
Good catch! Will be fixing this.

GohJW commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,225 @@
+        // Update the keepAlive flag
+        String[] commandList = splitIntoSubcommands(command);
+        String mainCommand = extractCommandFromSubcommand(commandList[0]);
+        switch (mainCommand) {
```
Good code formatting, allows for easy readability.

guillaume-grn commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,15 @@
+
+    @Override
+    public String getDescription() {
+        String desc = "[D]" + super.getDescription();
```
desc is not super explicit

nichyjt replied:
Noted, I agree that this should more descriptive

guillaume-grn commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,225 @@
+        System.out.println("────────────────────────────────────────────────────────────");
+    }
+
+    public static void Exit() {
```
camelCase standard : it should be "exit"
```suggestion
    public static void exit() {
```

javienneyeo commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,225 @@
+        return command.split(" --");
+    }
+
+    public static HashMap<String, String> getSubcommandMap(String[] commandList) {
```
good that the methods are named in a way that it is easy to know what they do

javienneyeo commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,225 @@
+        return subcommandText;
+    }
+
+    public static Task createTask(HashMap<String, String> commandMap, String taskType) {
```
this method is a little long, exceeding more than 30 lines of code

nichyjt replied:
Hmm, its long due to the presence of guard clauses for error handling. Perhaps the error checking could be refactored out into a function -- but I think that this may cause issues down the road with complexity.

I'll experiment to see what works best!

javienneyeo commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,225 @@
+        }
+        return newTask;
+    }
+
```
good that the code is easily readable

guillaume-grn commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,225 @@
+        }
+    }
+
+    public static void Greet() {
```
Same as above
```suggestion
    public static void greet() {
```

javienneyeo commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,225 @@
+        String[] commandList = splitIntoSubcommands(command);
+        String mainCommand = extractCommandFromSubcommand(commandList[0]);
+        switch (mainCommand) {
+        case "Bye":
```
good that the methods and variables are named properly that clearly shows the intent of it

guillaume-grn commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,29 @@
+        return "[" + this.getStatusIcon() + "] " + this.description;
+    }
+
+    public boolean checkIsDone() {
```
Suggestion : Maybe you could adapt the method name because boolean methods should be named to sound like booleans.

guillaume-grn reviewed (commented):
Overall the standards are respected and the format is also very good.

nichyjt committed:
Change code to adhere to camelCase standard (PR comment)

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Adhere to boolean naming convention (PR comment)

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add command parsing logic to reduce nest level in the event driver "Duke.java" file.

Defer "long method logic" by suggestion by @javienneyeo to Level-5 increment. Abstracting out error handling will reduce logic size.

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add Command Enum to fix magic number issue.

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Merge branch 'branch-PR-Fixes'

nichyjt committed:
Fix minor naming nit

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add a command validator to check user input correctness

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add a custom exception class to indicate bad user command input

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Fix minor argument length bugs to adhere to CommandParser logic of accounting main command + sub commands

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Rename method names to more accurately describe that command parsing includes the main command

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Refactor handleCommand:
1. Abstract out case specific logic into their own methods (Resolves PR suggestion)
2. Add try-catch checks for both command and case-specific logic

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Edit EXPECTED.TXT to fit the new output strings due to exception handling

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Merge branch 'branch-Level-5'

- Add exception handlers
- Add Command enum
- Add Command validator

nichyjt committed:
Remove redundant error throws and add in comments stating assertions/invariants, if any

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Fix to follow code standards

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Update documentation for correctness

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Remove redundant static keyword

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Refactor source files into packages

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Update source path

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add delete command

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Refactor task handling into a TaskManager Controller

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Refactor task management logic

TaskManager now handles all task-related operations rather than the main file
Use ArrayList instead

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add taskmanager-specific exception

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add data handler class

DataHandler takes care of read/write of task data to disk

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add DataError class

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Refactor Tasks to get various params separately

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add debug mode and save/load function

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add debug command

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Merge branch 'branch-Level-6'

nichyjt committed:
Merge Level-7

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Bugfix: Add debug mode

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add save data folder to gitignore

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Update regression tests

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

qianz-z commented on src/main/java/max/Max.java:
```diff
@@ -0,0 +1,141 @@
+
+    private static TaskManager taskManager;
+
+    public static void setIsListening(boolean keepListening) {
```
Try to make your boolean variable name sound like boolean :>

qianz-z commented on src/main/java/max/command/CommandParser.java:
```diff
@@ -0,0 +1,78 @@
+        // For example, the deadline subcommand: "by Friday, 13th"
+        // Should return "by"
+        String[] words = subcommand.split(" ");
+        return words[0];
```
Try to avoid magic literals. There has been a few more instances in the rest of your code. You can use a named constant makes the code easier to understand because the name tells us more about the meaning of the number.

qianz-z commented on src/main/java/max/data/PersistentDataHandler.java:
```diff
@@ -0,0 +1,225 @@
+    private static final String delimiter = " -- ";
+
+    // These constants define the index position of task arguments in the token array
+    private static final int TASK_TYPE_INDEX = 0;
```
Good use of constants to avoid magic numbers! Keep this up for the rest of your code!

qianz-z commented on src/main/java/max/data/PersistentDataHandler.java:
```diff
@@ -0,0 +1,225 @@
+            }
+        }
+        return isDataExist;
+    }
+    private String tokenizeTasks(ArrayList<Task> arrayList) {
```
Remember to keep a line spacing between methods!

qianz-z commented on src/main/java/max/data/PersistentDataHandler.java:
```diff
@@ -0,0 +1,225 @@
+                newTask = new Todo(taskDescription);
+                break;
+            case "D":
+                try {
```
Be careful of deep nestings!

qianz-z commented on src/main/java/max/task/TaskManager.java:
```diff
@@ -0,0 +1,119 @@
+        int taskNum;
+        try {
+            taskNum = convertTaskNumberString(taskNumString);
+        }catch (TaskException exception){
```
I think you may have missed out a space before `catch`. Do check the rest of your code for similar violations.

qianz-z reviewed (commented):
Overall, your iP looks good! Try to be consistent in creating named variables to avoid magic literals! Keep up the good effort!

nichyjt committed:
Fix formatting nits

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add UI class

UI class abstracts out terminal printing.
Utilities such as notifying errors are added.

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Refactor to use Ui class

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Rename persistent data to Storage for consistency

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Update regression test

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add dateparser

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add date format checking to TaskManager

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Update regression tests

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Format code to standards

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add find, fetch commands

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add find function and reduce magic literals

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add find, fetch commands to Max event driver

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Update regression tests

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Fix Command docstring format

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add JavaDoc

Explain clearly how the user input format works

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Refactor magic literals and Add JavaDoc

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Remove unused file

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Refactor out magic literals

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Remove depreciated DataError enum

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add JavaDoc for Storage

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Update JavaDoc with format naming consistency

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add JavaDoc and refactor out magic literals

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Refactor out magic literals

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Refactor out magic literals

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add JavaDoc and add utility methods

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Rectify JavaDoc formatting nits

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Remove magic literals and add JavaDoc

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add JavaDoc

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add JavaDoc and refactor magic literals

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Fix formatting nits

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Fix formatting nits

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add JavaDoc to Class headers

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Merge pull request #2 from nichyjt/branch-Level-8

Merge Level-8 date features

nichyjt committed:
Merge branch 'master' into branch-Level-9

nichyjt committed:
Resolve merge conflicts

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Merge pull request #1 from nichyjt/branch-Level-9

Merge Level-9 Find task feature

nichyjt committed:
Resolve merge conflicts

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add JavaDoc for merged Date file

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Fix formatting nits

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Fix output inconsistencies

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Update regression test

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Merge pull request #3 from nichyjt/branch-A-JavaDoc

1. Merge JavaDoc additions
2. Fix inconsistencies
3. Update regression tests

nichyjt committed:
Update docs

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add TOC for User Guide and minor text fixes

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Rename package to adhere to standards

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Fix double greet bug

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Update regression test

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Remove magic int literals

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add optional documentation

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Fix format nit

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Refactor long methods

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Refactor logic building logic

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Refactor long methods out

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Refactor long methods out

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add method to build logo

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Refactor out long detokenize method

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Update regression test

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Update formatting nit

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Remove redundant comment

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Refactor unused function and buildLogo

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Update Javadoc

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Update user guide

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add instructions on starting Max

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Add private access modifier to DataDirectory

Signed-off-by: nichyjt <nicholas.yek@gmail.com>

nichyjt committed:
Refactor to use OS-agnostic newline

Signed-off-by: nichyjt <nicholas.yek@gmail.com>
